airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/command/DestinationStream.kt

```diff
@@ -59,6 +59,8 @@ data class DestinationStream(
     val isFileBased: Boolean = false,
     // whether we will move the file (in addition to the metadata)
     val includeFiles: Boolean = false,
+    val destinationObjectName: String? = null,
+    val matchingKey: List<String> = emptyList<String>(),
```

Comment:
nit: given that this is optional, I think nullable is better than default to empty list.

Reply:
I wouldn't attach any expectations w.r.t data activation.
I am more thinking, if it was defined in the input, then it's not null, otherwise null. If it is a data activation object with an operation that requires it and it's missing, then it's fine to fail explicitly.